distutils/file_util.py

```diff
@@ -140,12 +140,13 @@ def copy_file(  # noqa: C901
         if not (os.path.exists(dst) and os.path.samefile(src, dst)):
             try:
                 os.link(src, dst)
-                return (dst, 1)
             except OSError:
                 # If hard linking fails, fall back on copying file
                 # (some special filesystems don't support hard linking
                 #  even under Unix, see issue #8876).
                 pass
+            else:
+                return (dst, 1)
```

Comment:
Honestly, I feel like TRY is overreaching here. There's no way that `return (dst, 1)` is going to cause an OSError, and re-writing with the else clause just distracts from the typical control flow.

Reply:
In general, this rule might help identify cases where it's not clear which part of the code may throw. The usefulness of the `TRY300` rule depends on the context: number of lines and complexity of the code under `try`.

In this specific case, it is quite clear that the `return` line won't raise en exception. On the other hand, I do not find the `else` clause less readable. The typical control flow is important of course, but so is clarity about what is tested under `try`. Additionally, the error control flow feels important too, enough to have 3 lines of comments.